### PR TITLE
WIP release 16.1.0

### PR DESCRIPTION
**Pull request by rlepinski**

### CODE REVIEW

File: android/src/main/java/com/urbanairship/reactnative/AirshipModule.kt [hunk 1]
```diff
@@ -27,6 +25,8 @@ class AirshipModule internal constructor(val context: ReactApplicationContext) :
     private var isOverrideForegroundDisplayEnabled: Boolean = false
     private val foregroundDisplayRequestMap = mutableMapOf<String, CompletableDeferred<Boolean>>()
 
+    private val scope: CoroutineScope = CoroutineScope(Dispatchers.Main) + SupervisorJob()
```
Comment (jyaganeh) [inline android/src/main/java/com/urbanairship/reactnative/AirshipModule.kt:28]: Is there a lifecycle callback we can use to cancel this scope? I think we'd want it to be on destroy of the application (not the hosting Activity)

**Review by jyaganeh — APPROVED** [review]
lgtm overall, just one question about the `scope` in `AirshipModule`

File: android/src/main/java/com/urbanairship/reactnative/AirshipModule.kt (hunk 1 above)
Comment (rlepinski) [reply android/src/main/java/com/urbanairship/reactnative/AirshipModule.kt:28]: I think modules outlive the activity but i could be wrong. Should we just make it a static var?

Comment (Ulrico972) [reply android/src/main/java/com/urbanairship/reactnative/AirshipModule.kt:28]: Hmm I think there is no way to listen for when the application itself is destroyed 🤔

**Review by Ulrico972 — APPROVED** [review]

**Pull request merged**